src/log.ts

```diff
@@ -809,25 +809,39 @@ class Log implements LogSeverityFunctions {
     const options = opts ? (opts as WriteOptions) : {};
     const self = this;
 
-    if (options.resource) {
-      if (options.resource.labels) {
-        options.resource.labels = snakeCaseKeys(options.resource.labels);
-      }
-      return writeWithResource(options.resource);
-    } else if (this.logging.detectedResource) {
-      return writeWithResource(this.logging.detectedResource);
-    } else {
-      const resource = await getDefaultResource(this.logging.auth);
-      this.logging.detectedResource = resource;
-      return writeWithResource(resource);
+    let decoratedEntries;
+    try {
+      decoratedEntries = self.decorateEntries_(arrify(entry) as Entry[]);
+    } catch (err) {
+      // Ignore errors (the API will speak up if it has an issue).
     }
-    async function writeWithResource(resource: {} | null) {
-      let decoratedEntries;
+
+    const allEntriesHaveResources = decoratedEntries.every(r => {
```

Comment:
I like this approach though, might still be worth considering skipping the metadata check?

Reply:
If `allEntriesHaveResources` turns out to be true, we skip the metadata check below with:

```js
if (allEntriesHaveResources) {
  return writeWithResource(null);
}
```

Is that what you meant?